lib/common/logging.c

```diff
@@ -177,10 +177,135 @@ set_format_string(int method, const char *daemon)
 
 #define DEFAULT_LOG_FILE CRM_LOG_DIR "/pacemaker.log"
 
+static bool
+logfile_disabled(const char *filename)
+{
+    return pcmk__str_eq(filename, "none", pcmk__str_casei|pcmk__str_null_matches)
+           || pcmk__str_eq(filename, "/dev/null", pcmk__str_none);
+}
+
+/*!
+ * \internal
+ * \brief Fix log file ownership if group is wrong or doesn't have access
+ *
+ * \param[in] filename  Log file name (for logging only)
+ * \param[in] logfd     Log file descriptor
+ *
+ * \return Standard Pacemaker return code
+ */
+static int
+chown_logfile(const char *filename, int logfd)
+{
+    uid_t pcmk_uid = 0;
+    gid_t pcmk_gid = 0;
+    struct stat st;
+
+    // Get the log file's current ownership and permissions
+    if (fstat(logfd, &st) < 0) {
+        return errno;
+    }
+
+    // Any other errors don't prevent file from being used as log
+
+    if (pcmk_daemon_user(&pcmk_uid, &pcmk_gid) != pcmk_ok) {
+        return pcmk_rc_ok;
```

Comment:
in which case would that not work?
shouldn't we have a similar warning as below as when we weren't able to figure out what to set then effectively we weren't able to set it properly.

Reply:
The possible errors are errno's from calloc() (ENOMEM) or getpwnam_r() (many including EIO, EPERM, etc., though unlikely -- the one time we've seen it is when the system disk was lost and the value hadn't been cached yet).

Good catch on the logging -- I thought pcmk_daemon_user() already logged failures, but it doesn't, so I'll add that.